rocBLAS configure failing to find OPENBLAS on Windows with -DBUILD_TESTING=ON

Configure and build for the `rocBLAS` CMake target succeeds with `-DBUILD_TESTING=OFF`. Fails with `-DBUILD_TESTING=ON`.

Logs:

```
[build] [12/16  75% :: 11.491] Configure sub-project rocBLAS (in background)
[build] FAILED: math-libs/BLAS/rocBLAS/stamp/configure.stamp math-libs/BLAS/rocBLAS/build/CMakeCache.txt math-libs/BLAS/rocBLAS/build/cmake_install.cmake compile_commands_fragment_rocBLAS.json D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/stamp/configure.stamp D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/CMakeCache.txt D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/cmake_install.cmake D:/projects/TheRock/build/compile_commands_fragment_rocBLAS.json 
[build] C:\Windows\system32\cmd.exe /C "cd /D D:\projects\TheRock\build\math-libs\BLAS\rocBLAS\build && d:\projects\TheRock\.venv\Scripts\python D:/projects/TheRock/build_tools/teatime.py --log-timestamps --label "rocBLAS configure" --interactive D:/projects/TheRock/build/logs/rocBLAS_configure.log -- "C:/Program Files/CMake/bin/cmake.exe" -GNinja -BD:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build -SD:/projects/TheRock/math-libs/BLAS/rocBLAS -DCMAKE_INSTALL_PREFIX=D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/stage -DTHEROCK_STAGE_INSTALL_ROOT=D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/stage -DCMAKE_TOOLCHAIN_FILE=D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/_toolchain.cmake -DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/_init.cmake -DHIP_PLATFORM=amd -DROCM_PATH= -DROCM_DIR= -DBUILD_WITH_TENSILE=OFF -DBUILD_WITH_HIPBLASLT=OFF -DTensile_TEST_LOCAL_PATH=D:/projects/TheRock/math-libs/BLAS/Tensile -DTENSILE_VERSION= -DBUILD_CLIENTS_TESTS=ON -DBUILD_CLIENTS_BENCHMARKS=ON -DLINK_BLIS=OFF && "C:\Program Files\CMake\bin\cmake.exe" -E touch D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/compile_commands.json && "C:\Program Files\CMake\bin\cmake.exe" -E touch D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/stamp/configure.stamp && "C:\Program Files\CMake\bin\cmake.exe" -E copy D:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/compile_commands.json D:/projects/TheRock/build/compile_commands_fragment_rocBLAS.json"
[build] [rocBLAS configure] -- Sub-project PKG_CONFIG_PATH: D:/projects/TheRock/build/third-party/sysdeps/windows/zlib/build/stage/lib/rocm_sysdeps/lib/pkgconfig;D:/projects/TheRock/build/third-party/sysdeps/windows/zstd/build/stage/lib/rocm_sysdeps/lib/pkgconfig;
[build] [rocBLAS configure] -- Resolving packages from super-project: ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx
[build] [rocBLAS configure] -- Augmented toolchain PATH=D:/projects/TheRock/build/core/clr/dist/bin;D:/projects/TheRock/build/core/clr/dist/lib/llvm/bin;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\bin\Roslyn;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\Extensions\Microsoft\CodeCoverage.Console;C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\\x64;C:\Program Files (x86)\Windows Kits\10\bin\\x64;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\\MSBuild\Current\Bin\amd64;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\Tools\;C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\Program Files\Python312\Scripts\;C:\Program Files\Python312\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;D:\tools\bin;C:\Program Files\Git\cmd;C:\Program Files\CMake\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\Users\Nod-Shark16\AppData\Local\Microsoft\WindowsApps;C:\Users\Nod-Shark16\AppData\Local\Programs\Microsoft VS Code\bin;c:\users\nod-shark16\.local\bin;c:\users\nod-shark16\appdata\roaming\python\python312\scripts;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\Llvm\x64\bin;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\VC\Linux\bin\ConnectionManagerExe
[build] [rocBLAS configure] -- Use amdclang to build for amdgpu backend
[build] [rocBLAS configure] -- Resolving system find_package(Threads) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] -- Resolving system find_package(Python3) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] -- Resolving super-project find_package(ROCM 0.7.3 QUIET BYPASS_PROVIDER NO_DEFAULT_PATH PATHS D:/projects/TheRock/build/base/rocm-cmake/dist/share/rocm/cmake)
[build] [rocBLAS configure] CMake Deprecation Warning at D:/projects/TheRock/build/base/rocm-cmake/dist/share/rocm/cmake/ROCMConfig.cmake:12 (message):
[build] [rocBLAS configure]   Use of find_package(ROCM) is deprecated as of ROCm 6.4.  Please use
[build] [rocBLAS configure]   find_package(ROCmCMakeBuildTools)
[build] [rocBLAS configure] Call Stack (most recent call first):
[build] [rocBLAS configure]   D:/projects/TheRock/cmake/therock_subproject_dep_provider.cmake:54 (find_package)
[build] [rocBLAS configure]   CMakeLists.txt:57 (find_package)
[build] [rocBLAS configure] 
[build] [rocBLAS configure] 
[build] [rocBLAS configure] -- Resolving super-project find_package(ROCmCMakeBuildTools BYPASS_PROVIDER NO_DEFAULT_PATH PATHS D:/projects/TheRock/build/base/rocm-cmake/dist/share/rocmcmakebuildtools/cmake)
[build] [rocBLAS configure] -- OS detected is unknown
[build] [rocBLAS configure] -- ROCM Platform Version:  is not set, using latest supported gfx list
[build] [rocBLAS configure] -- Using GPU_TARGETS: gfx1100;gfx1101;gfx1102
[build] [rocBLAS configure] -- Resolving super-project find_package(hip REQUIRED BYPASS_PROVIDER NO_DEFAULT_PATH PATHS D:/projects/TheRock/build/core/clr/dist/lib/cmake/hip)
[build] [rocBLAS configure] -- OS: unknown 
[build] [rocBLAS configure] -- Resolving system find_package(Threads) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] -- Resolving system find_package(Git) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] -- *** Building rocBLAS commit: e778093683a32d4e3e139f664aa289e55faac7db
[build] [rocBLAS configure] -- ** NOTE: blas2/rocblas_ger_kernels.cpp is compiled with the verbose flag -v for QC purposes.
[build] [rocBLAS configure] -- ** rocblas_library_settings: rocblas
[build] [rocBLAS configure] -- Resolving system find_package(Threads) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] -- Resolving system find_package(OpenMP) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] -- CLIENT COMMON CXX_OPTIONS: 
[build] [rocBLAS configure] -- CLIENT COMMON LINK: OpenMP::OpenMP_CXX;libomp;Threads::Threads
[build] [rocBLAS configure] -- Resolving system find_package(OPENBLAS) (not found in super-project ZLIB;zstd;AMDDeviceLibs;Clang;LLD;LLVM;amd_comgr;rocm-core;hip;HIP;hip-lang;hiprtc;hipblas-common;ROCmCMakeBuildTools;ROCM;GTest;msgpack-cxx)
[build] [rocBLAS configure] CMake Error at D:/projects/TheRock/cmake/therock_subproject_dep_provider.cmake:62 (find_package):
[build] [rocBLAS configure]   Could not find a package configuration file provided by "OPENBLAS" with any
[build] [rocBLAS configure]   of the following names:
[build] [rocBLAS configure] 
[build] [rocBLAS configure]     OPENBLASConfig.cmake
[build] [rocBLAS configure]     openblas-config.cmake
[build] [rocBLAS configure] 
[build] [rocBLAS configure]   Add the installation prefix of "OPENBLAS" to CMAKE_PREFIX_PATH or set
[build] [rocBLAS configure]   "OPENBLAS_DIR" to a directory containing one of the above files.  If
[build] [rocBLAS configure]   "OPENBLAS" provides a separate development package or SDK, be sure it has
[build] [rocBLAS configure]   been installed.
[build] [rocBLAS configure] Call Stack (most recent call first):
[build] [rocBLAS configure]   clients/CMakeLists.txt:176 (find_package)
[build] [rocBLAS configure] 
[build] [rocBLAS configure] 
[build] [rocBLAS configure] CMake Error at clients/CMakeLists.txt:176 (find_package):
[build] [rocBLAS configure]   Could not find a package configuration file provided by "OPENBLAS" with any
[build] [rocBLAS configure]   of the following names:
[build] [rocBLAS configure] 
[build] [rocBLAS configure]     OPENBLASConfig.cmake
[build] [rocBLAS configure]     openblas-config.cmake
[build] [rocBLAS configure] 
[build] [rocBLAS configure]   Add the installation prefix of "OPENBLAS" to CMAKE_PREFIX_PATH or set
[build] [rocBLAS configure]   "OPENBLAS_DIR" to a directory containing one of the above files.  If
[build] [rocBLAS configure]   "OPENBLAS" provides a separate development package or SDK, be sure it has
[build] [rocBLAS configure]   been installed.
[build] [rocBLAS configure] 
[build] [rocBLAS configure] 
[build] [rocBLAS configure] -- Configuring incomplete, errors occurred!
[build] [rocBLAS configure completed in 1 seconds]
```

Relevant sub-project code, with some `find_library` / `find_package` and hardcoded paths:
https://github.com/ROCm/rocBLAS/blob/d93e919f945e76ee89457c83d72aac4cb473f492/clients/CMakeLists.txt#L117-L181

I see https://github.com/ROCm/TheRock/blob/main/cmake/finders/FindBLAS.cmake already, with `OpenBLAS` package name. Maybe case sensitivity difference?

With both those fixes, now hitting `error: attempt to use a poisoned identifier` when building the rocBLAS tests on Windows:

```
115.0	[70/140  48% :: 115.010] Building CXX object clients/common/CMakeFiles/rocblas_clients_testing_common.dir/blas_ex/common_gemm_ex.cpp.obj
143.6	[70/140  49% :: 143.550] Generating ../staging/rocblas_gtest.data
148.5	[135/140  50% :: 148.456] Building CXX object clients/gtest/CMakeFiles/rocblas-test.dir/rocblas_test.cpp.obj
148.5	FAILED: clients/gtest/CMakeFiles/rocblas-test.dir/rocblas_test.cpp.obj 
148.5	ccache D:\projects\TheRock\build\core\clr\dist\lib\llvm\bin\clang++.exe -DCLIENTS_NO_FORTRAN -DGOOGLE_TEST -DROCBLAS_INTERNAL_API -DROCBLAS_NO_DEPRECATED_WARNINGS -DROCM_USE_FLOAT16 -DUSE_PROF_API=1 -D_SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING -D__HIP_PLATFORM_AMD__=1 -ID:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/include/rocblas -ID:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/include/rocblas/internal -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../include -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../include/blas1 -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../include/blas2 -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../include/blas3 -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../include/blas_ex -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/blas_ex -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../../library/include -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../../library/src/include -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/../../library/src -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/common -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/library/src -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/library/include -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/library/include/internal -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/library/src/include -ID:/projects/TheRock/build/math-libs/BLAS/rocBLAS/build/include -ID:/projects/TheRock/math-libs/BLAS/rocBLAS/library/src/blas3/Tensile -isystem D:/projects/TheRock/build/core/clr/dist/include -isystem D:/projects/TheRock/build/third-party/host-blas/dist/lib/host-math/include/openblas -isystem D:/projects/TheRock/build/third-party/googletest/dist/include -DWIN32 -D_CRT_SECURE_NO_WARNINGS -DNOMINMAX -fms-extensions -fms-compatibility -D_ENABLE_EXTENDED_ALIGNED_STORAGE  -Wno-documentation-unknown-command -Wno-documentation-pedantic -Wno-unused-command-line-argument -Wno-ignored-attributes -Wno-unknown-attributes -Wno-duplicate-decl-specifier --hip-path=D:/projects/TheRock/build/core/clr/dist --hip-device-lib-path=D:/projects/TheRock/build/core/clr/dist/lib/llvm/amdgcn/bitcode -D__HIP_HCC_COMPAT_MODE__=1 -O3 -DNDEBUG -std=c++17 -D_DLL -D_MT -Xclang --dependent-lib=msvcrt -Wno-unused-result -x hip --offload-arch=gfx1100 --offload-arch=gfx1101 --offload-arch=gfx1102 -fopenmp=libomp -MD -MT clients/gtest/CMakeFiles/rocblas-test.dir/rocblas_test.cpp.obj -MF clients\gtest\CMakeFiles\rocblas-test.dir\rocblas_test.cpp.obj.d -o clients/gtest/CMakeFiles/rocblas-test.dir/rocblas_test.cpp.obj -c D:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/rocblas_test.cpp
148.5	In file included from D:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/rocblas_test.cpp:30:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\windows.h:221:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\ole2.h:37:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\objbase.h:27:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\combaseapi.h:363:
148.5	C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\objidlbase.h:2319:14: error: attempt to use a poisoned identifier
148.5	 2319 |     FILETIME ctime;
148.5	      |              ^
148.5	In file included from D:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/rocblas_test.cpp:30:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\windows.h:221:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\ole2.h:37:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\objbase.h:28:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\coml2api.h:24:
148.5	C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\propidlbase.h:478:14: error: attempt to use a poisoned identifier
148.5	  478 |     FILETIME ctime;
148.5	      |              ^
148.5	In file included from D:/projects/TheRock/math-libs/BLAS/rocBLAS/clients/gtest/rocblas_test.cpp:30:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\windows.h:221:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\ole2.h:37:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\objbase.h:259:
148.5	In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\urlmon.h:439:
148.5	C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\msxml.h:1666:66: error: attempt to use a poisoned identifier
148.5	 1666 |         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE abort( void) = 0;
148.5	      |                                                                  ^
148.5	C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um\msxml.h:9057:66: error: attempt to use a poisoned identifier
161.2	 9057 |         virtual /* [helpstring][id] */ HRESULT STDMETHODCALLTYPE abort( void) = 0;

161.2	      |                                                                  ^

161.2	4 errors generated when compiling for gfx1100.
```

--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -172,7 +172,7 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
                     NO_DEFAULT_PATH
                   )
       if (NOT BLAS_LIBRARY)
-        find_package( OPENBLAS CONFIG REQUIRED )
+        find_package( OpenBLAS CONFIG REQUIRED )
         set( BLAS_LIBRARY OpenBLAS::OpenBLAS )
         set( BLAS_INCLUDE_DIR "" )
       endif()

--- a/clients/gtest/rocblas_test.cpp
+++ b/clients/gtest/rocblas_test.cpp
@@ -21,18 +21,23 @@
  * ************************************************************************ */
 
 #include "rocblas_test.hpp"
-#include "client_utility.hpp"
 
-#include <cstdlib>
-#include <exception>
-#include <regex>
 #ifdef WIN32
+// Must include windows.h before dependent headers.
+// Specifically this must be before client_utility.hpp uses `#pragma GCC poison`
+// to poison 'ctime' and 'abort', as those are used in standard library headers.
 #include <windows.h>
 #define strcasecmp(A, B) _stricmp(A, B)
 #else
 #include <pthread.h>
 #include <unistd.h>
 #endif
+
+#include "client_utility.hpp"
+
+#include <cstdlib>
+#include <exception>
+#include <regex>
 
 /*********************************************
  * thread pool functions

